cgsmiles/sample.py

```diff
@@ -10,17 +11,134 @@
+        probs = probs / sum(probs)
+        bonding = random.choices(bonds, weights=probs)[0]
+    else:
+        bonding  = random.choice(bonds)
+    return bonding
+
+def _set_bond_order_defaults(bonding_dict):
+    default_dict = {}
+    for bond_operator, prob in bonding_dict.items():
+        # we need to patch the bond order space
+        if not bond_operator[-1].isdigit():
+            bond_operator += '1'
+        default_dict[bond_operator] = prob
+    return default_dict
+
 class MoleculeSampler:
     """
     Given a fragment string in CGSmiles format and probabilities for residues
     to occur, return a random molecule with target molecular weight.
+
+    First, this class has to be initiated using the class construction
+    method `from_string`, which makes sure to read and resolve the fragment
+    graphs provided in the CGSmiles string.
+
+    Once the `MoleculeSampler` is initiated you can call the `sampler` method
+    in order to generate a new random polymer molecule from the fragment string
+    that was provided.
+
+    In addition to just randomly creating a molecule reactivities and termination
+    probabilities can be provided to steer the outcome. The sampler features a
+    two level connectivity determination. First using the `polymer_reactivities`
+    an open bonding descriptor from the growing polymer molecule is selected
+    according to the probabilities provided. Subsequently, a fragment is randomly
+    chosen that has a matching complementary bonding descriptor.
+
+    If in addition the `fragment_reactivities` are provided the algorithm will
+    select a fragment that features a bonding descriptor according to the
+    conditional probabilities provided. The `fragment_reactivities` keyword
+    accepts a two level dict. The first key is the bonding descriptor found on
+    the polymer molecule (i.e. the one that will be used to extend the polymer).
+    The second level contains a dict specifying a probability for the bonding
+    descriptors found in the fragments (i.e. their conditional probability given
+    the polymer descriptors).
+
+    Basic Examples
+    --------------
+    Random-copolymer of PMMA and PS with equal probabilities.
+
+    >>> cgsmiles_str = "{#PMMA=[>]C(C)C[<]C(=O)OC,#PS=[>]CC[<]c1cccc1}"
+    >>> sampler = MoleculeSampler.from_string(cgsmiles_str)
+    >>> polymer_graph = sampler.sample(target_weight=100000)
+
+    One can also make a blocky copolymer but steer the probability of having
+    head-to-head tail-to-tail vs head-to-tail addition. This is done by labeling
+    the bonding descriptors A-D and providing the probability that a certain
+    bonding descriptor is used to extend the polymer.
+
+    >>> cgsmiles_str = "{#PMMA=[>A]C(C)C[<B]C(=O)OC,#PS=[>C]CC[<D]c1cccc1}"
+    >>> polymer_reactivites = {">A":0.4, "<B":0.1, ">C": 0.4, "<D": 0.1}
+    >>> sampler = MoleculeSampler.from_string(cgsmiles_str)
+    >>> polymer_graph = sampler.sample(target_weight=100000)
```

Comment:
I'm not sure I understand this example. Once I have a PMMA with a free `<B`, isn't it just as likely to take `>A` (=PMMA) as it is `>C` (=PS)?
Besides a block copolymer, could you add an example to create an alternating copolymer (using the chances)

Also, whitespace after the `:` in `">A":0.4, "<B":0.1`

Reply:
Yep the example was somehow budge. I corrected it and moved it to the bottom of the basic examples